src/schoology-extractor/schoology_extractor/api/request_client_base.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Created this base class for not having circular dependencies between RequestClient and PaginatedResult:
Without the base class:
RequestClient returns PaginatedResult # In order to annotate this, I needed to import PaginatedResult  inside of RequestClient
PaginatedResult needs a RequestClient # For get_next_page method

With the base class:
RequestClient(RequestClientBase) # Imports only RequestClientBase
PaginatedResult needs a RequestClientBase # Only needs access to base get method(not get_users or any other method)

Reply:
interesting, haven't run into this dependency issue yet, but i can see it happening.

Reply:
@stephenfuqua @lalrg  The more I look at this the less I like it.  Fortunately I found an answer:
[Python type hints with imported class methods
](https://stackoverflow.com/questions/63234592/python-type-hints-with-imported-class-methods)
Basically you avoid this by guarding the type-only import with a "TYPE_CHECKING" constant:
```
from __future__ import annotations
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .request_client import RequestClient
```

Reply:
TypeScript solves this with the "import type" variation of import.